File: user-service/src/main/java/com/vb/springboot/user/model/request/CreateUserRequest.java
package com.vb.springboot.user.model.request;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

public class CreateUserRequest {

    @NotNull(message = "First name cannot be null")
    @Size(min = 2, message = "First name must not be less than two characters")
    private String firstName;

    @NotNull
    private String lastName;

    @Email
    private String email;

    @NotNull
    private String password;

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }
}

File: user-service/src/main/kotlin/com/vb/springboot/user/model/request/CreateUserRequest.kt
package com.vb.springboot.user.model.request

import javax.validation.constraints.Email
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

data class CreateUserRequest(
        @field:NotNull(message = "First name cannot be null")
        @field:Size(min = 2, message = "First name must not be less than two characters")
        var firstName: String?,

        @field:NotNull
        var lastName: String?,

        @field:Email
        var email: String?,

        @field:NotNull
        var password: String?,
)

